packages/sdk/src/sync-agent/user/models/metadata/displayName.ts

```diff
@@ -0,0 +1,117 @@
+import { check, dlogger } from '@river-build/dlog'
+import { LoadPriority, type Identifiable, type Store } from '../../../../store/store'
+import {
+    PersistedObservable,
+    persistedObservable,
+} from '../../../../observable/persistedObservable'
+import type { RiverConnection } from '../../../river-connection/riverConnection'
+import type { Client } from '../../../../client'
+import { isDefined } from '../../../../check'
+import type { IStreamStateView } from '../../../../streamStateView'
+import { make_MemberPayload_DisplayName } from '../../../../types'
+const logger = dlogger('csb:userSettings')
+
+type DisplayNameModel = {
+    isEncrypted: boolean
+    displayName: string
+}
+
+export interface UserMetadata_DisplayNameModel extends Identifiable {
+    id: string
+    streamId: string
```

Comment:
We need a way to let the client define the rules for the user metadata storage.

Towns for example doesn't care about channel specific display names, they only store display names inside space and dm streams, and not in channel streams. Would be nice to allow those rules.

Reply:
later we figure it out